### PR TITLE
Public apis promisified

### LINKED ISSUE #812: Switch from callbacks to promises
We should use promises rather than callbacks in parts of our API that require asynchronity:

* `Viewer#importXML`
* `Viewer#saveXML`
* `Viewer#saveSVG`
* `Moddle#fromXML`
* `Moddle#toXML`

Why switch:

* interact nicely with ES6 `async/await`
* easier consumption for the user

__Tasks__

* [x] Decide for transition strategy
* [x] Roll out change (https://github.com/bpmn-io/bpmn-js/issues/1311)
* [ ] Update documentation (https://github.com/bpmn-io/bpmn-js-examples/issues/112)
* [ ] Blog about it (https://github.com/bpmn-io/bpmn-js/issues/1320)
* [x] :heart: IE11

### PR DESCRIPTION
**Pull request by oguzeroglu**
Closes #812

Ticket for moddle APIs: https://github.com/bpmn-io/bpmn-moddle/issues/71

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "*.css"
   ],
   "devDependencies": {
+    "babel-polyfill": "^6.26.0",
```
Comment (nikku) [inline package.json]: Can we only Polyfill `Promise` please? We do not want to blindly polyfill any current and future ES improvement (yet).

File: lib/util/PromisifierUtil.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import { isFunction, some } from 'min-dash';
+
+export function promisifyIfNecessary(api) {
+
+  return function() {
+
+    // @oguz:
```
Comment (nikku) [inline lib/util/PromisifierUtil.js]: Isn't the function always the last argument? At least that is common convention in callback style APIs.

File: lib/util/PromisifierUtil.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      );
+
+      return api.apply(this, arguments);
+    }
+
+    if (!window.Promise) {
+      throw new Error('Promises are not supported for this browser. Consider polyfilling Promises.');
+    }
+
+    // eslint-disable-next-line no-undef
```
Comment (nikku) [inline lib/util/PromisifierUtil.js]: Declare a global `Promise` variable in eslint to get rid of the warning instead.

File: test/spec/ViewerSpec.js [hunk 4]
```diff
@@ -798,9 +800,15 @@ describe('Viewer', function() {
         try {
           viewer.importXML(xml);
```
Comment (nikku) [inline test/spec/ViewerSpec.js]: Must await this.

Comment (oguzeroglu) [reply test/spec/ViewerSpec.js]: I don't get why we want to await this. In this test we want to see if viewer.importXML throws an exception if we don't pass a callback and Promises are not polyfilled.

**Comment by oguzeroglu** [comment]
We'll change our approach on this topic.

**Pull request closed without merging**